presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java

```diff
@@ -316,6 +334,32 @@ private static void includeOrcColumnsRecursive(List<OrcType> types, Set<Integer>
         return Optional.of(ImmutableMap.copyOf(fields));
     }
 
+    private static Map<Integer, Slice> createIntermediateKeysMap(
+            Map<Integer, Slice> columnsToKeys,
+            Map<Integer, Integer> dwrfEncryptionGroupMap,
+            OrcDataSourceId dataSourceId)
+    {
+        Map<Integer, Slice> intermediateKeys = new HashMap<>(dwrfEncryptionGroupMap.values().size());
+        for (Map.Entry<Integer, Slice> entry : columnsToKeys.entrySet()) {
+            Slice key = entry.getValue();
+            int group = dwrfEncryptionGroupMap.get(entry.getKey());
+            Slice previous = intermediateKeys.putIfAbsent(group, key);
+            if (previous != null && !key.equals(previous)) {
+                throw new OrcCorruptionException(dataSourceId, "intermediate keys mapping does not match encryption groups");
```

Comment:
nit: maybe ass key and previous in the error message?

Reply:
I don't want to log unencrypted keys.